x-pack/legacy/plugins/actions/server/builtin_action_types/webhook.ts

```diff
@@ -4,31 +4,103 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { i18n } from '@kbn/i18n';
+import { URL } from 'url';
+import Boom from 'boom';
+import { Option, fromNullable } from 'fp-ts/lib/Option';
 import axios, { AxiosError, AxiosResponse } from 'axios';
 import { schema, TypeOf } from '@kbn/config-schema';
 import { getRetryAfterIntervalFromHeaders } from './lib/http_rersponse_retry_header';
 import { nullableType } from './lib/nullable';
 import { isOk, promiseResult, Result } from './lib/result_type';
-import { ActionType, ActionTypeExecutorOptions, ActionTypeExecutorResult } from '../types';
+import {
+  ConfigureableActionType,
+  ActionTypeExecutorOptions,
+  ActionTypeExecutorResult,
+  ActionType,
+} from '../types';
 
 // config definition
 enum WebhookMethods {
   POST = 'post',
   PUT = 'put',
 }
 
-export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
-
 const HeadersSchema = schema.recordOf(schema.string(), schema.string());
-
-const ConfigSchema = schema.object({
+const configSchemaProps = {
   url: schema.string(),
   method: schema.oneOf([schema.literal(WebhookMethods.POST), schema.literal(WebhookMethods.PUT)], {
     defaultValue: WebhookMethods.POST,
   }),
   headers: nullableType(HeadersSchema),
+};
+export type ActionTypeConfigType = TypeOf<typeof UnvalidatedConfigSchema>;
```

Comment:
Hmm fair.
The reason for "ActionTypeConfigType" is to align with the other actions.
The UnvalidatedConfigSchema is because... it doesn't have validation attached, the reason being that we can't yet attach validation because we don't have the configuration at the time of parsing the module yet.

I was hoping to find a way to "attach" validation to a Schema and so we wouldn't need two versions but it doesn't look possible.

I'll rename it so it aligns, because I know what you mean... but... it also feel wrong to not be explicit about it not having the expected validation... :/